tools/integration_tests/readdirplus/setup_test.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/mounting/static_mounting"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/setup"
+	"github.com/stretchr/testify/require"
+)
+
+const testDirName = "dirForReaddirplusTest"
+
+var (
+	storageClient *storage.Client
+	ctx           context.Context
+	testDirPath   string
+	mountFunc     func([]string) error
+	// mount directory is where our tests run.
+	mountDir string
+	// root directory is the directory to be unmounted.
+	rootDir string
+)
+
+func loadLogLines(reader io.Reader) ([]string, error) {
+	content, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(string(content), "\n"), nil
+}
+
+func validateLogsForReaddirplus(t *testing.T, logFile string, dentryCacheEnabled bool, startTime, endTime time.Time) {
+	t.Helper()
+
+	logForReadDirPlus := "ReadDirPlus ("
+	logForReadDir := "ReadDir ("
+	logForLookUpInode := "LookUpInode ("
+
+	file, err := os.Open(logFile)
+	require.NoError(t, err, "Failed to open log file")
+	defer file.Close()
+
+	logLines, err := loadLogLines(file)
+	require.NoError(t, err, "Failed to read log file")
+
+	foundReadDirPlus := false
+	foundReadDir := false
+	foundLookUpInode := false
+	for _, line := range logLines {
+		logEntry, err := read_logs.ParseJsonLogLineIntoLogEntryStruct(line) // Assuming read_logs can parse general log lines too or a more generic parser is available.
+		// If parsing fails, it might be a non-JSON line or a different structured log.
+		// For this specific message, we expect it to be in the "Message" field of a structured log.
+
+		if err == nil && logEntry != nil {
+			// Check if the log entry's timestamp is within the expected window.
+			if (logEntry.Timestamp.After(startTime) || logEntry.Timestamp.Equal(startTime)) &&
+				(logEntry.Timestamp.Before(endTime) || logEntry.Timestamp.Equal(endTime)) {
+				if strings.Contains(logEntry.Message, logForReadDirPlus) {
+					foundReadDirPlus = true
+				}
+				if strings.Contains(logEntry.Message, logForReadDir) {
+					foundReadDir = true
+				}
+				if strings.Contains(logEntry.Message, logForLookUpInode) {
+					foundLookUpInode = true
+				}
+			}
+		}
+	}
+
+	require.True(t, foundReadDirPlus, "ReadDirPlus not called")
+	require.False(t, foundReadDir, "ReadDir called unexpectedly")
+	if dentryCacheEnabled {
+		require.False(t, foundLookUpInode, "LookUpInode called unexpectedly")
+	} else {
+		require.True(t, foundLookUpInode, "LookUpInode not called")
+	}
+}
+
+func mountGCSFuseAndSetupTestDir(t *testing.T, flags []string, testDirName string) {
+	t.Helper()
+
+	// When tests are running in GKE environment, use the mounted directory provided as test flag.
+	if setup.MountedDirectory() != "" {
+		mountDir = setup.MountedDirectory()
+	}
+	setup.MountGCSFuseWithGivenMountFunc(flags, mountFunc)
+	setup.SetMntDir(mountDir)
+	testDirPath = setup.SetupTestDirectory(testDirName)
+}
+
+func TestMain(m *testing.M) {
```

Comment:
Please also add the entry in the - https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/run_tests_mounted_directory.sh